php/libraries/DefaultFilesUploadHandler.class.inc

```diff
@@ -0,0 +1,367 @@
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class DefaultFilesUploadHandler implements FilesUploadHandler
+{
+    /**
+     * Default directory to move the uploaded files to if none is explicitely
+     * specified.
+     */
+    public static $DEFAULT_MOVE_DIR = __DIR__ . "../user_uploads";
+    /**
+     *  The LorisForm used to upload the file(s).
+     */
+    private $_lorisForm;
+    /**
+     * Directory to move the uploaded files to. Will be created if it does not
+     * exist.
+     */
+    private $_moveDirPath;
+    /**
+     * When creating $_moveDirPath, these are the access rights to give it
+     * (should be an octal number).
+     */
+    private $_dirPermissions;
+    /**
+     * When creating $_moveDirPath, give it this access group.
+     */
+    private $_dirGroup;
+    /**
+     * All the errors that occured either when the files are validated, moved to
+     * their final locations or processed.
+     */
+    private $_errors;
+    /**
+     * The files and their final locations (if successfully moved). Each value
+     * in this array is a SplFileInfo. Each key is the name of the LorisForm
+     * field used to upload the file.
+     */
+    private $_movedFiles;
+    /**
+     * All the properties of the uploaded files. Equivalent to the $_FILES
+     * superglobal.
+     */
+    private $_files = null;
+
+    /**
+     * Builds a new handler for uploaded files.
+     *
+     * @param LorisForm $lorisForm      the LorisForm used to upload the file(s).
+     * @param int       $dirPermissions if directory $moveDirPath (see below) has
+     *                                  to be created, it will have these access
+     *                                  rights (should be an octal number).
+     * @param string    $moveDirPath    where the files should be moved once they are
+     *                                  uploaded. If this directory is specified and
+     *                                  does not exist, it will be created
+     *                                  automatically. If null, then the uploaded
+     *                                  files will be moved to $DEFAULT_MOVE_DIR.
+     * @param int       $dirGroup       if directory $moveDirPath has to be created
+     *                                  it will have this access group.
+     *
+     * @return nothing.
+     */
+    public function __construct(
+        LorisForm $lorisForm,
+        int $dirPermissions,
+        string $moveDirPath=null,
+        string $dirGroup=null
+    ) {
+        $this->_lorisForm      = $lorisForm;
+        $this->_moveDirPath    = is_null($moveDirPath)
+            ? self::$DEFAULT_MOVE_DIR : $moveDirPath;
+        $this->_dirPermissions = $dirPermissions;
+        $this->_dirGroup       = $dirGroup;
+
+        $this->_errors = array();
+    }
+
+    /**
+     * Adds an error (any kind) to the set of current errors.
```

Comment:
I don't know what a "kind" of error means in this context, so I find this parenthesis confusing.